### DISCUSSION TITLE
loss cloneNode method

### DISCUSSION THREAD

**Discussion by StringKe** [alpinejs/alpine#3711] [7. Bugs (archived)] [answered]
The morph plugin requires the execution of the `window.Alpine.cloneNode`

The API marked as `INTERNAL` seems to have been removed after compilation

https://github.com/alpinejs/alpine/blob/main/packages/alpinejs/src/alpine.js#L72

https://github.com/alpinejs/alpine/blob/2e520c199d4514a6c2e95011dd2bf1b79795eab4/packages/morph/src/morph.js#L42

![CleanShot 2023-08-18 at 17 07 28@2x](https://github.com/alpinejs/alpine/assets/31089228/5aca5a0b-8c9a-429e-8603-ccf57b68b5d1)

https://unpkg.com/browse/alpinejs@3.12.3/dist/cdn.js cloneNode not exists

link1: https://laracasts.com/discuss/channels/livewire/uncaught-typeerror-windowalpineclonenode-is-not-a-function

**Comment by joshhanley** [comment]
@StringKe that only happens if you're running Livewire V3 and still have an old version of Alpine running somewhere. You should disable Alpine, as per the selected answer on that Laracasts discussion, as Livewire V3 now has Alpine built in, which currently build from the main Alpine branch (not the latest Alpine release), as it's still in beta.

**Reply by StringKe** [reply]
Understood. But even without livewire, the `window.Alpine.cloneNode` is executed in the morph plugin, which is not correct.

**Reply by joshhanley** [reply]
@StringKe yep, the changes haven't been tagged yet. So make sure to only use morph 3.12.3 and Alpine 3.12.3.

**Comment by StringKe** [comment, marked as answer]
resolve https://github.com/livewire/livewire/discussions/6238#discussioncomment-6759716
